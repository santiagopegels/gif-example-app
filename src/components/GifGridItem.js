import React from 'react'

export const GifGridItem = ({ id, title, url }) => {
    return (
        <div>
            <img src={url} />
            <p>{title}</p>
        </div>
    )
}
