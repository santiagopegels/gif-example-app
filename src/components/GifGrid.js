import React from 'react'
import { GifGridItem } from './GifGridItem';
import { useFetchGif } from '../hooks/useFetchGif';

export const GifGrid = ({ category }) => {

    const { data: images, loading } = useFetchGif(category);

    return (
        <>
            <h3>{category}</h3>

            {loading && <p>Loading...</p>}

            <div className="card-grid">
                {
                    images.map((img) => (
                        <GifGridItem
                            key={img.id}
                            {...img}
                        />
                    ))
                }
            </div>
        </>
    )
}
