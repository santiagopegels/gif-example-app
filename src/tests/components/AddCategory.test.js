import React from 'react'
import { shallow } from "enzyme"
import { AddCategory } from "../../components/AddCategory"

describe('AddCategory component test', () => {

    const setCategories = () => {}
    const wrapper =  shallow(<AddCategory setCategories={setCategories} />)

    test('should show it correctly', () => {

        expect( wrapper ).toMatchSnapshot()        
    
    })
    
    test('should change the textbox', () => {
        const input = wrapper.find('input')
        const value = 'Hola mundo'

        input.simulate('change', {
            target:
            {
                value
            }
        });


    })
    

})
