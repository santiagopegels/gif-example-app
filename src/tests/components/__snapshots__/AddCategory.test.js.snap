// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddCategory component test should show it correctly 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
